src/Enqueue/Blocks/EnqueueBlocks.php

```diff
@@ -35,7 +35,7 @@ class EnqueueBlocks extends AbstractEnqueueBlocks
 	 *
 	 * @var ValidatorInterface
 	 */
-	public $validator;
+	protected $validator;
```

Comment:
Why `protected` and not `private`? Is this class meant to be a parent class?

Reply:
I have tried but got 
`PHP Fatal error:  Uncaught Error: Cannot access private property EightshiftForms\Rest\Routes\FormSubmitActiveCampaignRoute::$validator in ~/wp-content/plugins/eightshift-forms/src/Rest/Routes/AbstractBaseRoute.php:134`

so no bueno :(

Reply:
Oh, I kinda see (well not related to this concrete class, but to all the classes extending the `AbstractBaseRoute` class). 

You are referencing `$this->validator->validate` inside the abstract class, so you expect every class that extends it to have this property. 

The problem that I see (a huge code smell) is that **this is not explicitly stated anywhere.**

In other words, were you to just extend this base class, and not implement a validator, you'd get a fatal error. 

The correct architectural solution would be to have this class implement a `ValidatorInterface`, with a method that will get the validator. Then every class extending it must implement the `getValidator()` method that you can then reference in your code.

So this part:

```php
		// Validate Params.
		$validate = $this->validator->validate($params, $files, $formId, $formData);
		if (!empty($validate)) {
			throw new UnverifiedRequestException(
				\esc_html__('Missing one or more required parameters to process the request.', 'eightshift-forms'),
				$validate
			);
		}
```

Would then be

```php
		// Validate Params.
		$validate = getValidator()->validate($params, $files, $formId, $formData);
		if (!empty($validate)) {
			throw new UnverifiedRequestException(
				\esc_html__('Missing one or more required parameters to process the request.', 'eightshift-forms'),
				$validate
			);
		}
```

This gives less ambiguity to what `$this->validator` is and comes from, and more flexibility in the validator implementation 🙂 

Once this is done I think the above suggestion about the visibility shouldn't cause an issue (especially in the enqueue block class, which has nothing to do with the `AbstractBaseRoute` class).

Reply:
awesome explenation I will make a issue for this to be fixed. Tnx I learned something today ❤️

Reply:
here it is https://github.com/infinum/eightshift-forms/issues/253